app/src/main/java/org/fossasia/openevent/app/core/attendee/list/AttendeesFragment.java

```diff
@@ -161,13 +163,14 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         binding = DataBindingUtil.inflate(inflater, R.layout.fragment_attendees, container, false);
-        binding.fabScanQr.setOnClickListener(v -> {
-            Intent scanQr = new Intent(context, ScanQRActivity.class);
-            scanQr.putExtra(MainActivity.EVENT_KEY, eventId);
-            startActivity(scanQr);
-        });
-
         binding.fabScanQr.getDrawable().setColorFilter(getResources().getColor(android.R.color.white), PorterDuff.Mode.SRC_ATOP);
+
+        if (featureDecider.shouldScanQR()) {
+            binding.fabScanQr.setOnClickListener(v -> featureDecider.openScanQRActivity(getActivity(), eventId));
+        } else {
+            binding.fabScanQr.setOnClickListener(v -> Toast.makeText(context, "Scanning is not available in this build", Toast.LENGTH_SHORT).show());
+        }
```

Comment:
This can be done via single method, `openScanQRActivity`

The `should...` method was needed in OSS version because we needed to hide the option

If `should...` method is used, then it needs to hide the scan FAB and other things, but it'd be an overkill since we'll be adding the feature in later versions nevertheless